### PR TITLE
docs(core): cut the prose of the package by a quarter

### PR DESCRIPTION
**Pull request by vboussot**
The docstrings and comments of `konfai` lose 3046 lines, a quarter of the package's prose, with no code change at all.

## What went

Justifications ("because", "which is why"), history and legacy ("used to", "previously", "before this change"), narratives of rejected alternatives, prose proofs, hedges, second explanations of a fact already stated, and the measurement stories and benchmark tables that set no constant. Three examples of the kind:

- `sources.py`, `_default_num_workers`: the benchmark table ("Measured on a quiet 24-core host, three cases, whole PREDICTION runs, 0.24 s / 0.34 s (2.5-D, patch [1, 192, 192])"). The rule that sets the worker count stays.
- `workflow.py`: the incident ("the Synthesis example shipped `Standardize(mask: None)` in training against `Standardize(mask: MASK)` here, and paid 409 HU of MAE instead of 98"). The warning's contract and the `check_training_transforms` key stay.
- `reduction.py`, `Vote.__call__`: the sort benchmark ("a 128-row slab of six 512x512 uint8 members peaked 1920 MiB sorted, 256 MiB here"). The tie-break contract and the CPU/CUDA divergence it exists for stay.

## What stayed

What a module, class or function is, in one sentence. Every constraint a caller must respect: axis order and units, the `transform_shape` exactness, the `patch_locality` kinds, the reduction contract and its layout, the graph contract, `state_dict` not recursing into nested networks, the config binder's rules, each backend's read granularity, the patch ordering invariant, what each refusal refuses. Every config key name. The `Args:` and `Returns:` blocks the documentation site renders. A measured number wherever a constant right beside it rests on it.

## How the code was held still

`prose_check.py` parses every file at the base revision and here, removes the docstrings from both, and compares the trees: any change to a statement, a signature, an import or a string that is not a docstring fails it. It reports `files whose code changed: 0`.

## Gates

`pixi run test-fast` all passed, mypy clean on 157 files, `ruff check` and `ruff format --check` clean, the strict docs build at zero warnings (the docstrings feed the API pages).

## Summary by CodeRabbit

* **Documentation**
  * Clarified documentation and inline comments across workflows, data processing, transformations, patching, datasets, training, prediction, evaluation, networking, and runtime utilities.
  * Improved explanations of configuration, streaming, memory budgeting, geometry, caching, checkpoint handling, distributed execution, and supported data formats.
  * Documented byte-order normalization during OME-Zarr reads.
  * No user-facing runtime behavior or public API signatures changed.

### CODE REVIEW

File: konfai/data/materialize.py [hunk 1]
```diff
@@ -124,12 +114,10 @@ def materialize(
...
-        The streamed path sweeps every unsatisfied :class:`Save` slab by slab; when the plan refuses
-        (a WHOLE_VOLUME stage, a destination without region writes, a failed sweep) the whole-volume
-        load writes the same caches, same bytes, more memory. ``allow_fallback=False`` raises instead
-        of loading; ``prefer_whole`` is the plan's LOAD choice (no fallback, no refusal);
-        ``rewrite=True`` recomputes the case and renames over the old entries (``--overwrite``). A
-        chain whose caches all exist writes nothing (the per-case resume).
+        The streamed path sweeps every unsatisfied :class:`Save` slab by slab, the whole-volume load
+        writes the same caches with more memory. ``allow_fallback=False`` raises instead of loading;
+        ``prefer_whole`` is the plan's LOAD choice; ``rewrite=True`` recomputes the case and renames
+        over the old entries (``--overwrite``). A chain whose caches all exist writes nothing.
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/data/materialize.py:121]: Fixed: the docstring now says allow_fallback=False raises when streaming is unavailable, and that a planned prefer_whole route still loads.

File: konfai/data/patching/manager.py [hunk 2]
```diff
@@ -381,10 +376,9 @@ def _load_augmentation_group(self, start_index: int, data_augmentations: DataAug
             return
 
         # The case tensor itself, once per copy. A draw hands back a fresh tensor or a view of what
-        # it was given and writes nothing into it (Foreign clones for a class that might), so a
-        # copy a draw did not select IS the case, as copy 0 already is. A clone per copy was a
-        # memcpy of the case dropped unread: 640 MiB and 0.22 s for 10 copies of a 64 MiB case,
-        # per group, per case, per epoch under inline augmentation (measured).
+        # it was given and writes nothing into it (Foreign clones for a class that might), so a copy
+        # a draw did not select IS the case. A clone per copy was 640 MiB and 0.22 s for 10 copies of
+        # a 64 MiB case, per group, per case, per epoch under inline augmentation.
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/data/patching/manager.py]: Fixed: the contract is stated, a draw must not write into the tensor it is handed (Foreign clones for a class that might).

File: konfai/network/network/network.py [hunk 3]
```diff
@@ -944,15 +905,12 @@ def downsampling_factor(self) -> list[int] | None:
...
-        evenly at every level, so the input must be a multiple of the coarsest downsampling the graph
-        reaches. That factor is traced through the branch register: a strided ``Conv`` or a ``MaxPool``
-        multiplies the branch it writes, while ``ConvTranspose``/``Upsample`` and a residual branch's
-        ``AvgPool`` pass through. Because the trace follows branches, a residual block's strided shortcut
-        (parallel to its strided main conv, merged by ``Add``) counts ONCE, not twice. Used to size a
-        free (``0``) patch axis to a valid extent (padded up, cropped back after the forward).
+        evenly at every level. The factor is traced through the branch register: a strided ``Conv`` or
+        ``MaxPool`` multiplies the branch it writes, ``ConvTranspose``/``Upsample`` and a residual
+        branch's ``AvgPool`` pass through, and a residual block's strided shortcut counts ONCE. Used to
+        size a free (``0``) patch axis to a valid extent (padded up, cropped back after the forward).
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/network/network/network.py:912]: Fixed, and it was wrong before this branch too: AvgPool multiplies the branch it writes like Conv and MaxPool; what keeps a residual shortcut's pool from inflating the factor is the per-axis maximum parallel paths merge by, which the docstring now says.

File: konfai/predictor/ensemble.py [hunk 4]
```diff
@@ -349,7 +328,7 @@ def forward(  # type: ignore[override]
             output_layers (list): List of output layer names to extract from each sub-model.
 
         Returns:
-            list[tuple[str, torch.Tensor]]: Aggregated output for each layer, after applying the reduction.
+            list[tuple[str, torch.Tensor]]: Aggregated output per layer, after the reduction.
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/predictor/ensemble.py]: Fixed: the Returns block names the three-element tuple, key, channel counts, tensor. The two-element form was already there on main.

File: konfai/utils/dicom.py [hunk 5]
```diff
@@ -16,30 +16,19 @@
...
+1. **Series discovery**: group files by SeriesInstanceUID; a folder may hold several series.
+2. **Slice ordering**: sort by ImagePositionPatient (z along the ImageOrientationPatient normal),
+   not by filename or InstanceNumber.
+3. **Geometry extraction**: spacing_mm (PixelSpacing plus the derived inter-slice distance), origin
+   (ImagePositionPatient of the first slice), direction cosines (ImageOrientationPatient rows and
+   columns plus their cross product for the z-axis).
+4. **CT intensity rescale**: RescaleSlope and RescaleIntercept convert stored pixel values to
+   Hounsfield Units, mandatory for CT and absent or identity for MR.
+5. **Error handling**: missing tags, single-slice series, inconsistent spacing, non-square pixels
+   and unsupported transfer syntaxes are reported.
```
[the remark this answers is not in the document]
Comment (vboussot) [reply konfai/utils/dicom.py]: Fixed: single-slice series and non-square pixels are removed from the error list; extract_geometry takes SliceThickness (or 1.0 mm) for a single slice and reads the two pixel spacings independently.
